hnn_core/viz.py

```diff
@@ -242,29 +248,45 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    from .dipole import Dipole
+    from .dipole import Dipole, average_dipoles
 
     if ax is None:
         _, ax = plt.subplots(1, 1, constrained_layout=True)
 
     if isinstance(dpl, Dipole):
         dpl = [dpl]
+        average = False
+        alpha = 1
```

Comment:
This all looks really good @chenghuzi! Sorry I'm late to the game. This is super minor, but I would avoid overwriting `alpha` here so that the user can still choose to set it's value if desired.

Reply:
Got it. just fixed that

Reply:
Actually, what happens if the user sets the `color` parameter but only supplies a single `Dipole` object to `plot_dipole`? IMO, they should be able set the color of individual trials if `average=False`, or the color of the average dipole if `average=True`.

Reply:
Once again, my only objective here is to avoid silently overwriting user-specified parameters to avoid confusion.

Reply:
Then the same parameter `color` has different meanings in that case, which I think may create even more confusion. In our current implementation, `color` is just for the averaged dpl line but not for individual dpls. As for the case of `average=False`, I don't think manually assigning different colors for different trials will be meaningful. Of course, a straightforward solution is to add another parameter like `individual_colors` as a list of color values.

Reply:
what one might want to do is display dipoles from two different conditions in the same figure by passing the `ax` parameter:

```py
fig = dpl_cond1.plot(color='r')
dpl_cond2.plot(color='b', ax=fig.gca())
```

I think it's fine to fix the color of the averaged dipole and trust that the user does not select this color? Alternatively if you want more control do something like:

```py
color : float | dict
   If dict, keys can be 'average' and 'trial'
```

But I think we can build this in later if there is a need. For now, I'd say YAGNI. Just use color for setting the color of individual trials

Reply:
^that works. There have also been times when I’ve needed to compute the trial-average dipole outside `plot_dipole` and then wished to plot it as a certain color within a preexisting matplotlib `axes` instance.

Reply:
A little bit confused. Are we going to stay with the current implementation? If we change to use `color` for individual trials, then a default color must be set as the matplotlib will assign different colors for each line if `color=None`. And to avoid conflicts, a default color for averaged line should also be set. And this brings us back to the `color : float | dict` solution I guess